lib/url.c

```diff
@@ -884,9 +885,6 @@ static void prune_dead_connections(struct Curl_easy *data)
                                  call_extract_if_dead)) {
       /* unlocked */
 
-      /* remove connection from cache */
-      Curl_conncache_remove_conn(data, prune.extracted, TRUE);
-
```

Comment:
How about leaving a comment saying that `extract_if_dead` already removed the conn from the cache? It saves the reader from having to look it up or think about that case.

Reply:
I have added a comment and did some more cleanup. Let me know if any of that is unwanted.